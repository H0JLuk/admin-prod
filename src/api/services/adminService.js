import { Api } from '../apiClient';
import { getReqOptions } from './index';
import _ from 'lodash';

export async function addUser(data) {
    return Api.post('/admin/user', data, getReqOptions());
}

const contentType = 'image/jpeg';
const options = {
    headers: {
        Accept: '*/*',
        'Content-Type': contentType
    }
};

const reqOptions = () => _.merge(options, getReqOptions());

export const getOffers = async () => {
    return Api.get('/admin/offer/excel', reqOptions(), 'blob');
};

export const getFeedback = async () => {
    return Api.get('/admin/feedback/excel', reqOptions(), 'blob');
};

export async function getBannerList() {
    return  Api.get('/banner/list', reqOptions());
}

export async function addBanner(bannerDto) {
    return Api.post('/admin/banner', { bannerDto }, reqOptions());
}

export async function deleteBanner(id) {
    return Api.delete(`/admin/banner/${id}`, reqOptions());
}

export async function uploadFile(binaryFile, path) {
    return Api.upload(`/admin/file?path=${path}`, binaryFile, reqOptions());
}
