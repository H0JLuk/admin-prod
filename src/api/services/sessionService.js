import { removeDefaultAppCode } from './clientAppService';

const TOKEN = 'dzo_token';
const ROLE = 'role';
const APP_CODE = 'appCode';

export function storeUserData(userToken, userRole) {
    window.sessionStorage.setItem(TOKEN, userToken);
    window.sessionStorage.setItem(ROLE, userRole);
}

export function deleteUserData() {
    window.sessionStorage.removeItem(TOKEN);
    window.sessionStorage.removeItem(ROLE);
    deleteAppCode();
    removeDefaultAppCode(); /* TODO: remove this after second phase */
}

export function getSession() {
    return window.sessionStorage.getItem(TOKEN);
}

export function getRole() {
    return window.sessionStorage.getItem(ROLE);
}

export function saveAppCode(appCode) {
    window.sessionStorage.setItem(APP_CODE, appCode);
}

export function deleteAppCode() {
    window.sessionStorage.removeItem(APP_CODE);
}

export function getAppCode() {
    return window.sessionStorage.getItem(APP_CODE);
}
