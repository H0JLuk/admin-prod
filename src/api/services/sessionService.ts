import ROLES from '@constants/roles';
import { removeDefaultAppCode } from './clientAppService';

const TOKEN = 'dzo_token';
const ROLE = 'role';
const APP_CODE = 'appCode';

export function storeUserData(userToken: string, userRole: string) {
    window.sessionStorage.setItem(TOKEN, userToken);
    window.sessionStorage.setItem(ROLE, userRole);
}

export function deleteUserData() {
    window.sessionStorage.removeItem(TOKEN);
    window.sessionStorage.removeItem(ROLE);
    deleteAppCode();
    removeDefaultAppCode(); /* TODO: remove this after second phase */
}

export function getSession() {
    return window.sessionStorage.getItem(TOKEN);
}

export function getRole(): ROLES {
    return (window.sessionStorage.getItem(ROLE) || '') as ROLES;
}

export function saveAppCode(appCode: string) {
    window.sessionStorage.setItem(APP_CODE, appCode);
}

export function deleteAppCode() {
    window.sessionStorage.removeItem(APP_CODE);
}

export function getAppCode() {
    return window.sessionStorage.getItem(APP_CODE);
}
