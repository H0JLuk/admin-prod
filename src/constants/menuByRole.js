import ROLES from './roles';
import { ROUTE, ROUTE_ADMIN, ROUTE_AUDITOR, ROUTE_OWNER, ROUTE_USER_MANAGER } from './route';
import { menuItemLables as labels } from './lables';

const AUDITOR_MENU_ITEMS = [
    { label: labels.AUDIT, path: ROUTE_AUDITOR.AUDIT }
];

const USER_MANAGER_MENU_ITEMS = [
    { label: labels.USERS, path: ROUTE_USER_MANAGER.USERS }
];

const ADMIN_MENU_ITEMS = [
    { label: labels.DASHBOARD, path: ROUTE.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.USERS },
    { label: labels.FILES, path: ROUTE_ADMIN.FILES },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.LANDING, path: ROUTE_ADMIN.LANDING },
    { label: labels.CATEGORY, path: ROUTE_ADMIN.CATEGORY },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.PROMO_CAMPAIGN },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const ADMIN_MENU_FOR_POINT_APP = [
    { label: labels.DASHBOARD, path: ROUTE.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.USERS },
    { label: labels.FILES, path: ROUTE_ADMIN.FILES },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.PROMO_CAMPAIGN },
];

const PRODUCT_OWNER_MENU_ITEMS = [
    { label: labels.DASHBOARD, path: ROUTE.DASHBOARD },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.LANDING, path: ROUTE_OWNER.LANDING },
    { label: labels.CATEGORY, path: ROUTE_OWNER.CATEGORY },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.PROMO_CAMPAIGN },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const PRODUCT_OWNER_MENU_FOR_POINT_APP = [
    { label: labels.DASHBOARD, path: ROUTE.DASHBOARD },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.PROMO_CAMPAIGN },
];

export const resolveMenuItemsByRoleAndAppCode = (role, appCode) => {
    switch (role) {
        case ROLES.AUDITOR:
            return AUDITOR_MENU_ITEMS;
        case ROLES.ADMIN:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? ADMIN_MENU_FOR_POINT_APP
                : ADMIN_MENU_ITEMS;
        case ROLES.PRODUCT_OWNER:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? PRODUCT_OWNER_MENU_FOR_POINT_APP
                : PRODUCT_OWNER_MENU_ITEMS;
        case ROLES.USER_MANAGER:
            return USER_MANAGER_MENU_ITEMS;
        default: return [];
    }
};
