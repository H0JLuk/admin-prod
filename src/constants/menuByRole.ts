import ROLES from './roles';
import {
    BUSINESS_ROLE_PAGES,
    ROUTE_ADMIN,
    ROUTE_AUDITOR,
    ROUTE_OWNER,
    ROUTE_PARTNER,
    ROUTE_USER_MANAGER,
    SALE_POINT_PAGES,
    LOCATIONS_PAGES,
} from './route';
import { menuItemLabels as labels } from './labels';
import { getDefaultAppCode } from '@apiServices/clientAppService';

/* OLD DESIGN MENU */
const AUDITOR_MENU_ITEMS: MenuItem[] = [
    { label: labels.AUDIT, path: ROUTE_AUDITOR.AUDIT },
];

const USER_MANAGER_MENU_ITEMS: MenuItem[] = [
    { label: labels.USERS, path: ROUTE_USER_MANAGER.USERS },
];

const OLD_ADMIN_MENU_ITEMS: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_ADMIN.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.OLD_DESIGN_USERS },
    { label: labels.REPORTS, path: ROUTE_ADMIN.REPORTS },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.PRESENTATION, path: ROUTE_ADMIN.PRESENTATION },
    { label: labels.CATEGORY, path: ROUTE_ADMIN.CATEGORY },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.OLD_DESIGN_PROMO_CAMPAIGN },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const OLD_ADMIN_MENU_FOR_POINT_APP: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_ADMIN.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.OLD_DESIGN_USERS },
    { label: labels.REPORTS, path: ROUTE_ADMIN.REPORTS },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.OLD_DESIGN_PROMO_CAMPAIGN },
];

const OLD_PRODUCT_OWNER_MENU_ITEMS: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_OWNER.DASHBOARD },
    { label: labels.REPORTS, path: ROUTE_OWNER.REPORTS },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.PRESENTATION, path: ROUTE_OWNER.PRESENTATION },
    { label: labels.CATEGORY, path: ROUTE_OWNER.CATEGORY },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.OLD_DESIGN_PROMO_CAMPAIGN },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const OLD_PRODUCT_OWNER_MENU_FOR_POINT_APP: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_OWNER.DASHBOARD },
    { label: labels.REPORTS, path: ROUTE_OWNER.REPORTS },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.OLD_DESIGN_PROMO_CAMPAIGN },
];

export const resolveOldMenuItemsByRoleAndAppCode = (role: ROLES, appCode?: string) => {
    if (!appCode) {
        appCode = getDefaultAppCode(); // TODO: remove getDefaultAppCode after second phase
    }

    switch (role) {
        case ROLES.AUDITOR:
            return AUDITOR_MENU_ITEMS;
        case ROLES.ADMIN:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? OLD_ADMIN_MENU_FOR_POINT_APP
                : OLD_ADMIN_MENU_ITEMS;
        case ROLES.PRODUCT_OWNER:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? OLD_PRODUCT_OWNER_MENU_FOR_POINT_APP
                : OLD_PRODUCT_OWNER_MENU_ITEMS;
        case ROLES.USER_MANAGER:
            return USER_MANAGER_MENU_ITEMS;
        default: return [];
    }
};
/* OLD DESIGN MENU */

type MenuItem = {
    label: string;
    path: string;
};

const USER_MANAGER_SIDEBAR_MENU_ITEMS: MenuItem[] = [
    { label: labels.USERS, path: ROUTE_USER_MANAGER.USERS },
];

const USER_MANAGER_TOP_MENU_ITEMS: MenuItem[] = [
];

const PARTNER_SIDEBAR_MENU_ITEMS: MenuItem[] = [
    { label: labels.USERS, path: ROUTE_PARTNER.USERS },
    { label: labels.APP, path: ROUTE_PARTNER.APPS },
    { label: labels.REPORTS, path: ROUTE_PARTNER.REPORTS },
];

const PARTNER_TOP_MENU_ITEMS: MenuItem[] = [
];

const ADMIN_SIDEBAR_MENU_ITEMS: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_ADMIN.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.USERS },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.REFERENCE_BOOKS, path: ROUTE_ADMIN.REFERENCE_BOOKS },
    { label: labels.CATEGORY, path: ROUTE_ADMIN.CATEGORY },
    { label: labels.CONSENTS, path: ROUTE_ADMIN.CONSENTS },
    { label: labels.APP, path: ROUTE_ADMIN.APPS },
];

const ADMIN_TOP_MENU_ITEMS: MenuItem[] = [
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.PROMO_CAMPAIGN },
    { label: labels.SETTINGS, path: ROUTE_ADMIN.APP_SETTINGS },
    { label: labels.PRESENTATION, path: ROUTE_ADMIN.PRESENTATION },
    { label: labels.GROUPS, path: ROUTE_ADMIN.GROUPS },
    { label: labels.REPORTS, path: ROUTE_ADMIN.REPORTS },
];

const ADMIN_REFERENCE_BOOKS_MENU_ITEMS = [
    { label: labels.SALE_POINT, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${SALE_POINT_PAGES.LIST}` },
    { label: labels.LOCATION, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${LOCATIONS_PAGES.LIST}` },
    { label: labels.BUSINESS_ROLE, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${BUSINESS_ROLE_PAGES.LIST}` },
    // { label: labels.ROLE_IN_CANAL, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${ROLE_IN_CANAL_PAGES.LIST}` },
    // { label: labels.DEVICE_TYPE, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${DEVICE_TYPE_PAGES.LIST}` },
    // { label: labels.CANAL_TYPE, path: `${ROUTE_ADMIN.REFERENCE_BOOKS}${CANAL_TYPE_PAGES.LIST}` },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const ADMIN_SIDEBAR_MENU_FOR_POINT_APP: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_ADMIN.DASHBOARD },
    { label: labels.USERS, path: ROUTE_ADMIN.USERS },
    { label: labels.DZO, path: ROUTE_ADMIN.DZO },
    { label: labels.REFERENCE_BOOKS, path: ROUTE_ADMIN.REFERENCE_BOOKS },
    { label: labels.CONSENTS, path: ROUTE_ADMIN.CONSENTS },
    { label: labels.APP, path: ROUTE_ADMIN.APPS },
];

const ADMIN_TOP_MENU_FOR_POINT_APP: MenuItem[] = [
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_ADMIN.PROMO_CAMPAIGN },
    { label: labels.SETTINGS, path: ROUTE_ADMIN.APP_SETTINGS },
    { label: labels.GROUPS, path: ROUTE_ADMIN.GROUPS },
    { label: labels.REPORTS, path: ROUTE_ADMIN.REPORTS },
];

const PRODUCT_OWNER_SIDEBAR_MENU_ITEMS: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_OWNER.DASHBOARD },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.CATEGORY, path: ROUTE_OWNER.CATEGORY },
    { label: labels.APP, path: ROUTE_OWNER.APPS },
];

const PRODUCT_OWNER_TOP_MENU_ITEMS: MenuItem[] = [
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.PROMO_CAMPAIGN },
    { label: labels.GROUPS, path: ROUTE_OWNER.GROUPS },
    { label: labels.PRESENTATION, path: ROUTE_OWNER.PRESENTATION },
    { label: labels.REPORTS, path: ROUTE_OWNER.REPORTS },
];

// Список доступных вкладок для приложений с кодом оканчивающимся на "-point", таких как "promo-point" и "mcdonalds-point"
const PRODUCT_OWNER_SIDEBAR_FOR_POINT_APP: MenuItem[] = [
    { label: labels.DASHBOARD, path: ROUTE_OWNER.DASHBOARD },
    { label: labels.DZO, path: ROUTE_OWNER.DZO },
    { label: labels.APP, path: ROUTE_OWNER.APPS },
];

const PRODUCT_OWNER_TOP_FOR_POINT_APP: MenuItem[] = [
    { label: labels.PROMO_CAMPAIGNS, path: ROUTE_OWNER.PROMO_CAMPAIGN },
    { label: labels.GROUPS, path: ROUTE_OWNER.GROUPS },
    { label: labels.REPORTS, path: ROUTE_OWNER.REPORTS },
];

export const resolveMenuItemsByRoleAndAppCode = (role: ROLES, appCode = '', isReferenceBookPage?: boolean) => {
    if (!appCode) {
        appCode = getDefaultAppCode(); // TODO: remove getDefaultAppCode after second phase
    }

    switch (role) {
        // case ROLES.AUDITOR:
        //     return AUDITOR_MENU_ITEMS;
        case ROLES.ADMIN:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? [
                    ADMIN_SIDEBAR_MENU_FOR_POINT_APP,
                    isReferenceBookPage ? ADMIN_REFERENCE_BOOKS_MENU_ITEMS : ADMIN_TOP_MENU_FOR_POINT_APP,
                ] : [
                    ADMIN_SIDEBAR_MENU_ITEMS,
                    isReferenceBookPage ? ADMIN_REFERENCE_BOOKS_MENU_ITEMS : ADMIN_TOP_MENU_ITEMS,
                ];
        case ROLES.PRODUCT_OWNER:
            return (appCode.endsWith('-point') && appCode !== 'greenday-point')
                ? [
                    PRODUCT_OWNER_SIDEBAR_FOR_POINT_APP,
                    isReferenceBookPage ? [] : PRODUCT_OWNER_TOP_FOR_POINT_APP,
                ] : [
                    PRODUCT_OWNER_SIDEBAR_MENU_ITEMS,
                    isReferenceBookPage ? [] : PRODUCT_OWNER_TOP_MENU_ITEMS,
                ];
        case ROLES.USER_MANAGER:
            return [USER_MANAGER_SIDEBAR_MENU_ITEMS, isReferenceBookPage ? [] : USER_MANAGER_TOP_MENU_ITEMS];
        case ROLES.PARTNER:
            return [PARTNER_SIDEBAR_MENU_ITEMS, isReferenceBookPage ? [] : PARTNER_TOP_MENU_ITEMS];
        default:
            return [[], []];
    }
};
