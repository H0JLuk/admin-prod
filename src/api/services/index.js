import { getSession } from './sessionService';

export const getReqOptions = () => {
    const token = getSession();
    return {
        headers: { token }
    };
};
