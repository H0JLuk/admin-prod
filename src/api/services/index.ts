import { ApiRequestHeaders, ApiRequestOptions } from '@types';
import { getDefaultAppCode } from './clientAppService';
import { getAppCode, getSession } from './sessionService';

const TOKEN_HEADER = 'token';
const APP_CODE_HEADER = 'clientAppCode';
const CONTENT_TYPE = 'Content-Type';
const ACCEPT = 'Accept';

export const getReqOptions = (contentType?: string, currAppCode?: string): ApiRequestOptions => {
    const token = getSession();
    const appCode = currAppCode || getAppCode() || getDefaultAppCode(); // TODO: remove getDefaultAppCode after second phase
    const headers = {} as ApiRequestHeaders;
    token && (headers[TOKEN_HEADER] = token);
    appCode && (headers[APP_CODE_HEADER] = appCode);
    contentType && (headers[ACCEPT] = '*/*');
    contentType && (headers[CONTENT_TYPE] = contentType);
    return { headers };
};
