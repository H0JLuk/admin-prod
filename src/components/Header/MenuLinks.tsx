import React from 'react';
import { NavLink, matchPath, useLocation } from 'react-router-dom';
import { getAppCode, getRole } from '@apiServices/sessionService';
import { resolveMenuItemsByRoleAndAppCode } from '@constants/menuByRole';
import { ROUTE_ADMIN } from '@constants/route';

import styles from './Header.module.css';

const routesForReferenceBooksMenu = [
    ROUTE_ADMIN.REFERENCE_BOOKS,
];

const MenuLinks = () => {
    const { pathname } = useLocation();
    const appCode = getAppCode() || '';
    const role = getRole();
    const [, menuItems] = resolveMenuItemsByRoleAndAppCode(role, appCode, !!matchPath(pathname, routesForReferenceBooksMenu));

    return (
        <div>
            {menuItems.map((item) => (
                <NavLink
                    key={item.label}
                    to={item.path}
                    className={styles.menuItem}
                    activeClassName={styles.active}
                >
                    {item.label}
                </NavLink>
            ))}
        </div>
    );
};

export default MenuLinks;
