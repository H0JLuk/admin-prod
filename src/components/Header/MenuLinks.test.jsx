import React from 'react';
import { mount } from 'enzyme';
import MenuLinks from './MenuLinks';
import { getAppCode, getRole } from '../../api/services/sessionService';
import { resolveMenuItemsByRoleAndAppCode } from '../../constants/menuByRole';
import { MemoryRouter } from 'react-router';

jest.mock('../../api/services/sessionService', () => ({
    getAppCode: jest.fn(),
    getRole: jest.fn()
}));

jest.mock('../../constants/menuByRole', () => ({
    resolveMenuItemsByRoleAndAppCode: jest.fn()
}));

const TEST_ITEM = {
    label: 'TEST_ITEM',
    path: 'TEST_PATH'
};

const TEST_RESOLVE_MENU_ITEMS = [
    TEST_ITEM.label,
    [
        { label: TEST_ITEM.label, path: TEST_ITEM.path },
    ]
];

describe('<MenuLinks /> test', () => {

    beforeEach(() => {
        resolveMenuItemsByRoleAndAppCode.mockReturnValue(TEST_RESOLVE_MENU_ITEMS);
    });

    it('should be mount snap', () => {
        expect(mount(
            <MemoryRouter>
                <MenuLinks />
            </MemoryRouter>
        ).html()).toMatchSnapshot();
    });

    it('should show MenuLinks', async () => {
        const Component = mount(
            <MemoryRouter>
                <MenuLinks />
            </MemoryRouter>
        );
        expect(getAppCode).toBeCalledTimes(1);
        expect(getRole).toBeCalledTimes(1);
        expect(Component.find('NavLink')).toHaveLength(1);
    });
});
