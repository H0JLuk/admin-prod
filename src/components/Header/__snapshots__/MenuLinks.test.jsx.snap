// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLinks /> test should be mount snap 1`] = `"<div><a class=\\"menuItem\\" href=\\"/TEST_PATH\\">TEST_ITEM</a></div>"`;
